.configurations/configuration.dsc.yaml

```diff
@@ -0,0 +1,29 @@
+# yaml-language-server: $schema=https://aka.ms/configuration-dsc-schema/0.2
+# Reference: https://github.com/microsoft/winget-create#building-the-client
+properties:
+  resources:
+    - resource: Microsoft.Windows.Developer/DeveloperMode
+      directives:
+        description: Enable Developer Mode
+        allowPrerelease: true
+      settings:
+        Ensure: Present
+    - resource: Microsoft.WinGet.DSC/WinGetPackage
+      id: vsPackage
+      directives:
+        description: Install Visual Studio 2022 (any edition is OK)
+        allowPrerelease: true
+      settings:
+        id: Microsoft.VisualStudio.2022.Community
+        source: winget
+    - resource: Microsoft.VisualStudio.DSC/VSComponents
+      dependsOn:
+        - vsPackage
+      directives:
+        description: Install required VS workloads
+        allowPrerelease: true
+      settings:
+        productId: Microsoft.VisualStudio.Product.Community
+        channelId: VisualStudio.17.Release
+        vsConfigFile: '${WinGetConfigRoot}\..\.vsconfig'
```

Comment:
There is another parameter you can use here to include recommended and optional components: Link for your reference: https://github.com/microsoft/VisualStudioDSC/blob/582d5fead3fcffb2796a193c7f357973798b7df4/Microsoft.VisualStudio.DSC/Microsoft.VisualStudio.DSC.psm1#L24-L29

I think you should only add `includedRecommended = true` here. We don't want to force users to include optional components if they decide to use this configuration file. It would be better to configure what is needed and let the user add the optional components as they see fit.